### PR TITLE
Fixing vtorc naming convention - database

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description
Cleanup references to master and slave in `vt/orchestrator` in the database that it uses

## Related Issue(s)
- #7110

## Checklist
- [ ] Should this PR be backported?
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/orchestrator/db/generate_patches.go [hunk 1]
```diff
@@ -513,7 +513,7 @@ var generateSQLPatches = []string{
 	`
 		ALTER TABLE
 			database_instance
-			ADD COLUMN master_uuid varchar(64) CHARACTER SET ascii NOT NULL AFTER oracle_gtid
+			ADD COLUMN primary_uuid varchar(64) CHARACTER SET ascii NOT NULL AFTER oracle_gtid
```
Comment (shlomi-noach) [inline go/vt/orchestrator/db/generate_patches.go:516]: Should be `source_uuid`: it's the `@@server_uuid` of the immediate _source_, not the topology's `primary`.

File: go/vt/orchestrator/db/generate_base.go [hunk 2]
```diff
@@ -28,21 +28,21 @@ var generateSQLBase = []string{
...
-          exec_master_log_pos bigint(20) unsigned NOT NULL,
-          seconds_behind_master bigint(20) unsigned DEFAULT NULL,
-          slave_lag_seconds bigint(20) unsigned DEFAULT NULL,
-          num_slave_hosts int(10) unsigned NOT NULL,
-          slave_hosts text CHARACTER SET ascii NOT NULL,
+          source_host varchar(128) CHARACTER SET ascii NOT NULL,
+          source_port smallint(5) unsigned NOT NULL,
+          replica_sql_running tinyint(3) unsigned NOT NULL,
+          replica_io_running tinyint(3) unsigned NOT NULL,
+          primary_log_file varchar(128) CHARACTER SET ascii NOT NULL,
```
Comment (shlomi-noach) [inline go/vt/orchestrator/db/generate_base.go:38]: this and the following should probably use the `source` terminology.

File: go/vt/orchestrator/db/generate_base.go [hunk 3]
```diff
@@ -28,21 +28,21 @@ var generateSQLBase = []string{
...
+          read_primary_log_pos bigint(20) unsigned NOT NULL,
+          relay_primary_log_file varchar(128) CHARACTER SET ascii NOT NULL,
+          exec_primary_log_pos bigint(20) unsigned NOT NULL,
+          seconds_behind_primary bigint(20) unsigned DEFAULT NULL,
```
Comment (shlomi-noach) [inline go/vt/orchestrator/db/generate_base.go:42]: `seconds_behind_source`

File: go/vt/orchestrator/db/generate_patches.go [hunk 4]
```diff
@@ -27,7 +27,7 @@ var generateSQLPatches = []string{
 	`
 		ALTER TABLE
 			database_instance
-			ADD COLUMN last_sql_error TEXT NOT NULL AFTER exec_master_log_pos
+			ADD COLUMN last_sql_error TEXT NOT NULL AFTER exec_primary_log_pos
```
Comment (shlomi-noach) [inline go/vt/orchestrator/db/generate_patches.go:30]: "source"

File: go/vt/orchestrator/db/generate_patches.go [hunk 5]
```diff
@@ -523,12 +523,12 @@ var generateSQLPatches = []string{
 	`
 		ALTER TABLE
 			database_instance
-			ADD COLUMN ancestry_uuid text CHARACTER SET ascii NOT NULL AFTER master_uuid
+			ADD COLUMN ancestry_uuid text CHARACTER SET ascii NOT NULL AFTER primary_uuid
```
Comment (shlomi-noach) [inline go/vt/orchestrator/db/generate_patches.go:526]: assuming you've changed the previous one, change this to `source_uuid`

File: go/vt/orchestrator/inst/instance_dao.go [hunk 6]
```diff
@@ -964,18 +964,18 @@ func readInstanceRow(m sqlutils.RowMap) *Instance {
...
-	instance.ReplicationIOThreadRuning = m.GetBool("slave_io_running")
+	instance.ReplicationSQLThreadRuning = m.GetBool("replica_sql_running")
+	instance.ReplicationIOThreadRuning = m.GetBool("replica_io_running")
 	instance.ReplicationSQLThreadState = ReplicationThreadState(m.GetInt("replication_sql_thread_state"))
 	instance.ReplicationIOThreadState = ReplicationThreadState(m.GetInt("replication_io_thread_state"))
 	instance.HasReplicationFilters = m.GetBool("has_replication_filters")
 	instance.SupportsOracleGTID = m.GetBool("supports_oracle_gtid")
 	instance.UsingOracleGTID = m.GetBool("oracle_gtid")
-	instance.SourceUUID = m.GetString("master_uuid")
+	instance.SourceUUID = m.GetString("primary_uuid")
```
Comment (shlomi-noach) [inline go/vt/orchestrator/inst/instance_dao.go:978]: `source_uuid`

File: go/vt/orchestrator/inst/instance_dao.go [hunk 7]
```diff
@@ -984,36 +984,36 @@ func readInstanceRow(m sqlutils.RowMap) *Instance {
 	instance.UsingMariaDBGTID = m.GetBool("mariadb_gtid")
 	instance.SelfBinlogCoordinates.LogFile = m.GetString("binary_log_file")
 	instance.SelfBinlogCoordinates.LogPos = m.GetInt64("binary_log_pos")
-	instance.ReadBinlogCoordinates.LogFile = m.GetString("master_log_file")
-	instance.ReadBinlogCoordinates.LogPos = m.GetInt64("read_master_log_pos")
-	instance.ExecBinlogCoordinates.LogFile = m.GetString("relay_master_log_file")
-	instance.ExecBinlogCoordinates.LogPos = m.GetInt64("exec_master_log_pos")
+	instance.ReadBinlogCoordinates.LogFile = m.GetString("primary_log_file")
```
Comment (shlomi-noach) [inline go/vt/orchestrator/inst/instance_dao.go:987]: this and the following should change to "source".

File: go/vt/orchestrator/inst/instance_dao.go [hunk 8]
```diff
@@ -984,36 +984,36 @@ func readInstanceRow(m sqlutils.RowMap) *Instance {
...
+	instance.ExecBinlogCoordinates.LogPos = m.GetInt64("exec_primary_log_pos")
 	instance.IsDetached, _ = instance.ExecBinlogCoordinates.ExtractDetachedCoordinates()
 	instance.RelaylogCoordinates.LogFile = m.GetString("relay_log_file")
 	instance.RelaylogCoordinates.LogPos = m.GetInt64("relay_log_pos")
 	instance.RelaylogCoordinates.Type = RelayLog
 	instance.LastSQLError = m.GetString("last_sql_error")
 	instance.LastIOError = m.GetString("last_io_error")
-	instance.SecondsBehindPrimary = m.GetNullInt64("seconds_behind_master")
-	instance.ReplicationLagSeconds = m.GetNullInt64("slave_lag_seconds")
+	instance.SecondsBehindPrimary = m.GetNullInt64("seconds_behind_primary")
```
Comment (shlomi-noach) [inline go/vt/orchestrator/inst/instance_dao.go:997]: "seconds_behind_source"

File: go/vt/orchestrator/inst/instance_dao.go [hunk 9]
```diff
@@ -1258,8 +1258,8 @@ func ReadProblemInstances(clusterName string) ([](*Instance), error) {
 				or (unix_timestamp() - unix_timestamp(last_checked) > ?)
 				or (replication_sql_thread_state not in (-1 ,1))
 				or (replication_io_thread_state not in (-1 ,1))
-				or (abs(cast(seconds_behind_master as signed) - cast(sql_delay as signed)) > ?)
-				or (abs(cast(slave_lag_seconds as signed) - cast(sql_delay as signed)) > ?)
+				or (abs(cast(seconds_behind_primary as signed) - cast(sql_delay as signed)) > ?)
```
Comment (shlomi-noach) [inline go/vt/orchestrator/inst/instance_dao.go:1261]: "source"

File: go/vt/orchestrator/inst/instance_dao.go [hunk 10]
```diff
@@ -2274,57 +2274,57 @@ func mkInsertOdkuForInstances(instances []*Instance, instanceWasActuallyFound bo
...
 		"gtid_purged",
 		"gtid_errant",
 		"mariadb_gtid",
 		"pseudo_gtid",
-		"master_log_file",
-		"read_master_log_pos",
-		"relay_master_log_file",
-		"exec_master_log_pos",
+		"primary_log_file",
+		"read_primary_log_pos",
```
Comment (shlomi-noach) [inline go/vt/orchestrator/inst/instance_dao.go:2298]: "source", here an in next variables

File: go/vt/orchestrator/inst/instance_dao.go [hunk 11]
```diff
@@ -2274,57 +2274,57 @@ func mkInsertOdkuForInstances(instances []*Instance, instanceWasActuallyFound bo
...
 		"gtid_mode",
 		"gtid_purged",
 		"gtid_errant",
 		"mariadb_gtid",
 		"pseudo_gtid",
-		"master_log_file",
-		"read_master_log_pos",
-		"relay_master_log_file",
-		"exec_master_log_pos",
+		"primary_log_file",
```
Comment (shlomi-noach) [inline go/vt/orchestrator/inst/instance_dao.go:2297]: "source"

File: go/vt/orchestrator/inst/instance_dao.go [hunk 12]
```diff
@@ -2274,57 +2274,57 @@ func mkInsertOdkuForInstances(instances []*Instance, instanceWasActuallyFound bo
...
+		"read_primary_log_pos",
+		"relay_primary_log_file",
+		"exec_primary_log_pos",
 		"relay_log_file",
 		"relay_log_pos",
 		"last_sql_error",
 		"last_io_error",
-		"seconds_behind_master",
-		"slave_lag_seconds",
+		"seconds_behind_primary",
```
Comment (shlomi-noach) [inline go/vt/orchestrator/inst/instance_dao.go:2305]: "source"

**Review by shlomi-noach — CHANGES_REQUESTED** [review]
Overall good, and this seems correct. However, it looks like we're now also making a distinction between `primary` and `source` (e.g. you've defined `source_host` and `source_port`).

This means we need to be careful what we call "primary" (the top, wirteable server) and what we call "source" (the immediate parent in the topology). In light of that, there's various other variables that need to be renamed "primary"->"source".

This is at your discretion, because it goes a bit sideways to the original intention of this PR; I'm mentioning this because you seem to have made the distinction in some places, so we may as well make it all the way through.

**Comment by GuptaManan100** [comment]
Made all the requested changes @shlomi-noach

File: go/vt/orchestrator/db/generate_base.go (hunk 3 above)
Comment (deepthi) [reply go/vt/orchestrator/db/generate_base.go]: you could also use replication_lag_seconds. That's what we ended up doing in all other places besides vtorc.

**Review by deepthi — APPROVED** [review]
I have one suggestion:
seconds_behind_source -> replication_lag_seconds.
The rest LGTM

**Pull request merged**